api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -139,9 +139,11 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 							Obs replacement = os.saveObs(o, changeMessage);
 							//The logic in saveObs evicts the old obs instance, so we need to update
 							//the collection with the newly loaded and voided instance
-							toRemove.add(o);
-							toAdd.add(os.getObs(o.getId()));
-							toAdd.add(replacement);
+							if(!replacement.equals(o)) {
```

Comment:
@wluyima So the question will be, whether parent obs has to be made dirty when a groupmember is added or removed?